### DISCUSSION TITLE
Android equivalent to popup windows?

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#10228] [Q&A]
My Windows-only app defined several windows (Windows-derived XAML and code behinds). A button or command from the first window would pop a second window.
What is the equivalent in Android? Instantiating a Window-derived class on Android throws an exception. Android uses activities instead I believe. How should one view/usercontrol/activity swap to another, such that the second view can be dismissed via the Back button on the phone or an action taken within the second view, to get back to the original view?

**Comment by lukaszossowski** [comment]
There are no windows in Android, it is a single view application. There are a couple of ways to achieve something like a popup. The thing I do is to create a special overlay control that sits on top of the main view and gets shown with proper content when necessary. 

You can capture the back button pressed and handle it appropriately.

**Reply by AArnott** [reply]
Thanks. The popup aspect isn't important to me. That's just how I did it in Windows. On Android, switching activities/views entirely would be fine and probably more natural. How would I do that?

**Reply by emmauss** [reply]
`Avalonia.Android.Avalonia` is a native android widget that acts as the host for the whole ui. You can add it to your own activities or fragments to host any avalonia ui. Certain platform features available with a normal avalonia activity will not be available when used this way, ie File system access from toplevel and back button handling. Those can be added to your custom activities by implementing the interfaces you see here, https://github.com/AvaloniaUI/Avalonia/blob/0145182f9159a2b3ca1033128526dbc9921f44c4/src/Android/Avalonia.Android/AvaloniaMainActivity.cs#L13 Transparency effects though will not be available.

You would have to also handle spawning and hosting your custom activities/fragments and navigation between them.

**Reply by maxkatz6** [reply]
Also, worth to mention, it's possible to show normal Popup on Android. Would be the easiet way.
